File: distinct.py
import pandas as pd

# Membuat dataframe dengan data yang lebih banyak
import numpy as np
np.random.seed(123)
n_rows = 1000
df = pd.DataFrame({
    'sku_name': np.random.choice(list('ABCDE'), n_rows),
    'category': np.random.choice(['X', 'Y', 'Z'], n_rows),
    'customer_id': np.random.randint(1, 201, n_rows)
})
print(df.shape)

# Menggunakan drop_duplicates bersama dengan groupby
tabel2 = df.drop_duplicates('customer_id')
print(tabel2.shape)
tabel2 = tabel2.groupby(['sku_name', 'category'])['customer_id'].nunique().reset_index(name='jumlah_pelanggan')

# Menampilkan hasil
print(tabel2)

File: distinct.r
library(dplyr)

set.seed(123)
n_rows <- 1000
df <- data.frame(
  sku_name = sample(LETTERS[1:5], n_rows, replace = TRUE),
  category = sample(c("X", "Y", "Z"), n_rows, replace = TRUE),
  customer_id = sample(1:200, n_rows, replace = TRUE)
)
print(dim(df))

# Menggunakan distinct bersama dengan group_by
tabel2 <- df %>% distinct(customer_id, .keep_all = TRUE)
print(dim(tabel2))

tabel2 <- tabel2 %>% group_by(sku_name, category) %>% summarise(jumlah_pelanggan = n_distinct(customer_id))

# Menampilkan hasil
print(tabel2)
